mindmeld/augmentation.py

```diff
@@ -0,0 +1,285 @@
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+
+        model_name = "tuner007/pegasus_paraphrase"
+        torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.tokenizer = PegasusTokenizer.from_pretrained(model_name)
+        self.model = PegasusForConditionalGeneration.from_pretrained(
+            model_name, force_download=False
+        ).to(torch_device)
+
+    def _get_response(self, query, num_return_sequences=10, num_beams=10):
+        """Generates paraphrase responses for given query.
+
+        Args:
+            query (str): An application query.
+            num_return_sequences (int, optional): Maximum number of paraphrases to be generated.
+            num_beams (int, optional):
+
+        Return:
+            paraphrases (list(str)): List of paraphrased queries.
+        """
+        batch = self.tokenizer.prepare_seq2seq_batch(
+            [query], truncation=True, padding="longest", max_length=60
+        )
+        translated = self.model.generate(
+            **batch,
+            max_length=60,
+            num_beams=num_beams,
+            num_return_sequences=num_return_sequences,
+            temperature=1.5,
+        )
+        paraphrases = self.tokenizer.batch_decode(translated, skip_special_tokens=True)
+        return paraphrases
+
+    def augment_queries(self, queries, **kwargs):
+        augmented_queries = []
+        for query in queries:
+            augmented_queries.extend(self._get_response(query, **kwargs))
+        return augmented_queries
+
+
+class MultiLingualParaphraser(Augmentor):
+    """Paraphraser class for generating paraphrases based on language code of the app
+    (for languages other than English).
+    """
+
+    def __init__(self, app_path, config=None):
+        """Initializes a multi-lingual paraphraser.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+
+        self.torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.language, _ = get_language_config(app_path=app_path)
+
+        en_model_name = "Helsinki-NLP/opus-mt-ROMANCE-en"
+        self.en_tokenizer = MarianTokenizer.from_pretrained(en_model_name)
+        self.en_model = MarianMTModel.from_pretrained(en_model_name)
+        self.en_model.to(self.torch_device)
+
+        target_model_name = "Helsinki-NLP/opus-mt-en-ROMANCE"
+        self.target_tokenizer = MarianTokenizer.from_pretrained(target_model_name)
+        self.target_model = MarianMTModel.from_pretrained(target_model_name).to(
+            self.torch_device
+        )
+
+    def _translate(self, **kwargs):
```

Comment:
enumerate kwargs

Reply:
The kwargs are variable here again (and a longer list). I was thinking of doing that in the docstring instead of explicitly enumerating them in the kwargs.

Reply:
So it looks like this function needs some compulsory arguments, like template, queries, models etc. There are two ways to do this:
a) Make the compulsory arguments position arguments with no defaults
b) Make these compulsory keyword args as follows:
```
def _translate(self, *, template, queries, model, tokenizer, **kwargs):
```

Reply:
Took approach b)